app/src/org/commcare/activities/CommCareWiFiDirectActivity.java

```diff
@@ -659,9 +730,9 @@ protected void deliverError(CommCareWiFiDirectActivity receiver, Exception e) {
             }
 
             @Override
-            protected void deliverResult(CommCareWiFiDirectActivity receiver, FormRecord[] result) {
-                if (result != null) {
-                    receiver.onZipSuccesful(result);
+            protected void deliverResult(CommCareWiFiDirectActivity receiver, Integer result) {
+                if (result == ZipTask.RESULT_SUCCESS) {

```

Comment:
This definitely needs to be done with intValue, Integer comparisons are unpredictable (sometimes they are cached/interned sometimes not, so it can work one day but not the nexst)